climada/trajectories/impact_calc_strat.py

```diff
@@ -0,0 +1,291 @@
+    Impact object.
+    """
+
+    @abstractmethod
+    def compute_impacts(
+        self,
+        exp: Exposures,
+        haz: Hazard,
+        vul: ImpactFuncSet,
+        risk_transf_attach: Optional[float] = None,
+        risk_transf_cover: Optional[float] = None,
+        calc_residual: bool = True,
+    ) -> Impact:
+        """
+        Calculates the total impact, including optional risk transfer application.
+
+        Parameters
+        ----------
+        exp : Exposures
+            The exposure data.
+        haz : Hazard
+            The hazard data (e.g., event intensity).
+        vul : ImpactFuncSet
+            The set of vulnerability functions.
+        risk_transf_attach : float, optional
+            The attachment point (deductible) for the global risk transfer mechanism.
+            If None, default to 0.
+        risk_transf_cover : float, optional
+            The cover limit for the risk transfer mechanism. If None, the cover
+            is assumed to be infinite (only the attachment applies).
+        calc_residual : bool, default=True
+            If True, the function returns the residual impact (after risk transfer).
+            If False, it returns the transferred impact (the part covered by the
+            risk transfer).
+
+        Returns
+        -------
+        Impact
+            An object containing the computed total impact matrix and metrics.
+
+        See Also
+        --------
+        ImpactCalcComputation : The default implementation of this interface.
+        """
+        ...
+
+
+class ImpactCalcComputation(ImpactComputationStrategy):
+    r"""
+    Default impact computation strategy using the core engine of climada.
+
+    This strategy first calculates the raw impact using the standard
+    :class:`ImpactCalc` logic and then applies a global risk transfer mechanism.
+
+    This risk transfer is distinct and non-exclusive with the risk transfer
+    defined through the Exposures object. While in the Exposures case, the transfer
+    is defined on a per coordinate basis, this one is applied total impact on the whole region
+    considered, and proportionally rescales the impacts per coordinate accordingly.
+
+    Notes
+    -----
+    The calculation is performed event-wise:
+
+    1. **Total Impact**: Calculate the total impact for each event
+       (sum of impacts across all exposure points).
+    2. **Transferred Risk per Event**: Defined as:
+       $$\min(\max(0, \text{Total Impact} - \text{attachement}), \text{cover})$$
+    3. **Residual Risk per Event**:
+       $$\text{Total Impact} - \text{Transferred Risk per Event}$$
+    4. **Adjustment**: The original impact per exposure point is scaled
+       by the ratio of (Residual Risk / Total Impact) or
+       (Transferred Risk / Total Impact) for that event.
+       This ensures the risk transfer is shared proportionally among all
+       impacted exposure points.
```

Comment:
I guess this is the scaling. Could you please explain what this corresponds too? I am not familiar with this risk transfer spreadind.